extensions/amp-render/1.0/amp-render.js

```diff
@@ -246,9 +247,49 @@ export class AmpRender extends BaseElement {
       this.mutateProps(
         dict({
           'render': (data) => {
+            if (this.element.getAttribute('binding') === 'no') {
+              return templates
+                .renderTemplateAsString(dev().assertElement(template), data)
+                .then((html) => dict({'__html': html}));
+            }
+
+            let elements;
+            if (this.element.getAttribute('binding') === 'refresh') {
+              return templates
+                .renderTemplate(dev().assertElement(template), data)
+                .then((el) => {
+                  elements = isArray(el) ? el : [el];
+                  return Services.bindForDocOrNull(this.element);
+                })
+                .then((bind) => {
+                  return bind.rescan(elements, [], {
+                    'fast': true,
+                    // bind.signals().get('FIRST_MUTATE') gives the timestamp (in ms) when mutation
+                    // occured, which is null for the initial render
+                    'update': bind.signals().get('FIRST_MUTATE') !== null,
+                  });
+                })
+                .then(() => {
+                  return dict({__html: elements[0].innerHTML}); // or outerHTML?
+                });
+            }
+
+            // binding = "always", default case
             return templates
-              .renderTemplateAsString(dev().assertElement(template), data)
-              .then((html) => dict({'__html': html}));
+              .renderTemplate(dev().assertElement(template), data)
```

Comment:
Could we do the `.renderTemplate` call once, and set up the rescan call in its resolved handler?

Reply:
done